core/src/test/scala/org/apache/spark/SparkFunSuite.scala

```diff
@@ -228,44 +229,62 @@ abstract class SparkFunSuite
    * appender and restores the log level if necessary.
    */
   protected def withLogAppender(
-      appender: Appender,
+      appender: AbstractAppender,
       loggerNames: Seq[String] = Seq.empty,
       level: Option[Level] = None)(
       f: => Unit): Unit = {
     val loggers = if (loggerNames.nonEmpty) {
-      loggerNames.map(Logger.getLogger)
+      loggerNames.map(LogManager.getLogger)
     } else {
-      Seq(Logger.getRootLogger)
+      Seq(LogManager.getRootLogger)
+    }
+    if (loggers.size == 0) {
+      throw new SparkException(s"Cannot get any logger to add the appender")
     }
     val restoreLevels = loggers.map(_.getLevel)
     loggers.foreach { logger =>
-      logger.addAppender(appender)
-      if (level.isDefined) {
-        logger.setLevel(level.get)
+      logger match {
+        case logger: Logger =>
+          logger.addAppender(appender)
+          appender.start()
+          if (level.isDefined) {
+            logger.setLevel(level.get)
+            logger.get().setLevel(level.get)
+          }
+        case _ =>
```

Comment:
Since `loggers` is a `Seq[Logger]`, is this branch ever reachable? Unless I'm mistaken, I think think we'll always hit the `case logger: Logger` case.